lib/clc/error.rb

```
@@ -0,0 +1,50 @@
+module Clc
+  module Error
+    # Custom error class
+    class Error < StandardError; end
+    # Raised when returns the HTTP status code 400
+    class BadRequest < Error; end
+    # Raised when returns the HTTP status code 401
+    class Unauthorized < Error; end
+    # Raised when returns the HTTP status code 403
+    class Forbidden < Error; end;
+    # Raised when returns the HTTP status code 404
+    class NotFound < Error; end
+    # Raised when returns the HTTP status code 429
+    class TooManyRequests < Error; end
+    # Raised when returns the HTTP status code 500
+    class InternalServerError < Error; end
+
+    class CustomErrors < Faraday::Response::Middleware
+      def on_complete(response)
+        case response[:status]
+        when 400
+          raise Clc::Error::BadRequest, error_message_400(response)
+        when 401
+          raise Clc::Error::Unauthorized, error_message_400(response)
+        when 403
+          raise Clc::Error::Forbidden, error_message_400(response)
+        when 404
+          raise Clc::Error::NotFound, error_message_400(response)
+        when 500
+          raise Clc::Error::InternalServerError, error_message_500(response)
+        end
```

We probably need some kind of `default` clause here. In case if API will return something strange from 400..600 range.
This code will silently pass in that case.